core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java

```diff
@@ -82,6 +80,26 @@ public ZeroCodeReportGeneratorImpl(ObjectMapper mapper) {
         this.mapper = mapper;
     }
 
+    /**
+     * gets unique steps from a scenario. In case of retries, the steps have same correlation id and if
+     * one of the retries is successful, we include it in the result(not the one which failed).
+     * @param steps
+     * @return
+     */
+    private List<ZeroCodeReportStep> getUniqueSteps(List<ZeroCodeReportStep> steps){
```

Comment:
Sounds correct. 👍 
Can you apply the same to the CSV results too.

The changes to be applied here:

> org.jsmart.zerocode.core.report.ZeroCodeReportGeneratorImpl#buildCsvRows

Reply:
@authorjapps  sure I added for csv too